packages/core/src/login/oidc/IClientRegistrar.test.ts

```diff
@@ -109,3 +109,19 @@ describe("handleRegistration", () => {
     expect(storageUtility.setForUser).toHaveBeenCalled();
   });
 });
+
+describe("determineSigningAlg", () => {
+  it("returns the preferred algorithm of the supported list", () => {
+    expect(
+      determineSigningAlg(["ES256", "HS256", "RS256"], ["ES256", "RS256"])
+    ).toBe("ES256");
+    expect(determineSigningAlg(["ES256", "HS256", "RS256"], ["RS256"])).toBe(
+      "RS256"
+    );
+    expect(determineSigningAlg(["RS256"], ["RS256"])).toBe("RS256");
+  });
+
+  it("returns undefined if there are no matches", () => {
```

Comment:
What if no preference is provided at all...? Needs an explicit test, just to communicate the behaviour (is it 'null' or first preferred entry...?)

Reply:
0cb556ce03d197a6ebb51075ed4de50e0a57d8d1